src/Symfony/Component/Mailer/Transport/FileTransportFactory.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\Transport;
+
+use Symfony\Component\Mailer\Exception\UnsupportedSchemeException;
+
+/**
+ * @author Hugo Alliaume <@kocal>
+ */
+final class FileTransportFactory extends AbstractTransportFactory
+{
+    public function create(Dsn $dsn): TransportInterface
+    {
+        if ('file' === $dsn->getScheme()) {
+            return new FileTransport($dsn->getOption('path'), $this->dispatcher, $this->logger);
+        }
+
+        throw new UnsupportedSchemeException($dsn);
+    }
+
+    public function supports(Dsn $dsn): bool
+    {
+        return 'null' === $dsn->getHost() && null !== $dsn->getOption('path');
```

Comment:
Better check only for file schema in `supports` method and throw `IncompleteDsnException` in `create` method if path is not provided.